src/backend/TrafficCourts/TrafficCourts.Coms.Client/ObjectManagementService.cs

```diff
@@ -354,7 +354,7 @@ private string GetFileName(IEnumerable<KeyValuePair<string, string>> metadata)
     private List<FileSearchResult> FilterSearchResults(List<FileSearchResult> results, FileSearchParameters parameters)
     {
         results = results
-            .Where(result => !TagsMatch(result, parameters) || !MetadataMatch(result, parameters))
+            .Where(result => TagsMatch(result, parameters) && MetadataMatch(result, parameters))
```

Comment:
This is so weird that it used to have !TagsMatch and !MetadataMatch before. Does not any make sense to me. Good catch!